context.go

```diff
@@ -992,6 +995,8 @@ func (c *Context) Header(key, value string) {
 
 // GetHeader returns value from request headers.
 func (c *Context) GetHeader(key string) string {
+	c.mu.RLock()
```

Comment:
`mu` bundles too many functions, even if there are no header-related operations. But as long as `mu` is used, locking will also occur in the logic related to `headers`. This will degrade performance.

Reply:
created a separate mutex for this

Reply:
LGTM

Reply:
this lock should add to requestHeader method.

func (c *Context) requestHeader(key string) string {
	c.hmu.RLock()
	defer c.hmu.RUnlock()
	return c.Request.Header.Get(key)
}

Reply:
moved lock inside